lib/PuppeteerSharp.Tests/PageTests/SetRequestInterceptionTests.cs

```diff
@@ -0,0 +1,360 @@
+﻿using Microsoft.AspNetCore.Http;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Text;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace PuppeteerSharp.Tests.PageTests
+{
+    [Collection("PuppeteerLoaderFixture collection")]
+    public class SetRequestInterceptionTests : PuppeteerPageBaseTest
+    {
+        [Fact]
+        public async Task ShouldIntercept()
+        {
+            await Page.SetRequestInterceptionAsync(true);
+            Page.RequestCreated += async (sender, e) =>
+            {
+                Assert.Contains("empty.html", e.Request.Url);
+                Assert.NotNull(e.Request.Headers);
+                Assert.Equal(HttpMethod.Get, e.Request.Method);
+                Assert.Null(e.Request.PostData);
+                Assert.Equal(ResourceType.Document, e.Request.ResourceType);
+                Assert.Equal(Page.MainFrame, e.Request.Frame);
+                Assert.Equal("about:blank", e.Request.Frame.Url);
+                await e.Request.ContinueAsync();
+            };
+            var response = await Page.GoToAsync(TestConstants.EmptyPage);
+            Assert.True(response.Ok);
+        }
+
+        [Fact]
+        public async Task ShouldStopIntercepting()
+        {
+            await Page.SetRequestInterceptionAsync(true);
+            async void EventHandler(object sender, RequestEventArgs e)
+            {
+                await e.Request.ContinueAsync();
+                Page.RequestCreated -= EventHandler;
+            }
+            Page.RequestCreated += EventHandler;
+            await Page.GoToAsync(TestConstants.EmptyPage);
+            await Page.SetRequestInterceptionAsync(false);
+            await Page.GoToAsync(TestConstants.EmptyPage);
```

Comment:
Let's see what do you think about this @Meir017.
I think this doesn't test what is trying to test, even on Puppeteer.
I would add a counter on the event. After line 45 that counter should be 1, after line 47 that counter should still be 1.

Make sense?

Reply:
makes sense, but this is how puppeteer implement it